### ISSUE TITLE
`misc-include-cleaner` false positive for `localtime_r`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#64587]
```cpp
#include <ctime>

void getLocaltime(struct tm &ltime)
{
    time_t t;
    time(&t);
    localtime_r(&t, &ltime);
}
```

```console
<source>:7:5: warning: no header providing "localtime_r" is directly included [misc-include-cleaner]
    2 | 
    3 | void getLocaltime(struct tm &ltime)
    4 | {
    5 |     time_t t;
    6 |     time(&t);
    7 |     localtime_r(&t, &ltime);
      |     ^
1 warning generated.
```
https://godbolt.org/z/Txehvrs8M

https://en.cppreference.com/w/c/chrono/localtime
